hudi-utilities/src/main/java/org/apache/hudi/utilities/sources/helpers/KafkaOffsetGen.java

```diff
@@ -175,93 +155,101 @@ public static long totalNewMessages(OffsetRange[] ranges) {
 
     private static final String KAFKA_TOPIC_NAME = "hoodie.deltastreamer.source.kafka.topic";
     private static final String MAX_EVENTS_FROM_KAFKA_SOURCE_PROP = "hoodie.deltastreamer.kafka.source.maxEvents";
-    private static final KafkaResetOffsetStrategies DEFAULT_AUTO_RESET_OFFSET = KafkaResetOffsetStrategies.LARGEST;
+    private static final KafkaResetOffsetStrategies DEFAULT_AUTO_RESET_OFFSET = KafkaResetOffsetStrategies.latest;
     public static final long DEFAULT_MAX_EVENTS_FROM_KAFKA_SOURCE = 5000000;
     public static long maxEventsFromKafkaSource = DEFAULT_MAX_EVENTS_FROM_KAFKA_SOURCE;
   }
 
-  private final HashMap<String, String> kafkaParams;
+  private final HashMap<String, Object> kafkaParams;
   private final TypedProperties props;
   protected final String topicName;
 
   public KafkaOffsetGen(TypedProperties props) {
     this.props = props;
-    kafkaParams = new HashMap<String, String>();
+    kafkaParams = new HashMap<String, Object>();
     for (Object prop : props.keySet()) {
       kafkaParams.put(prop.toString(), props.getString(prop.toString()));
     }
     DataSourceUtils.checkRequiredProperties(props, Collections.singletonList(Config.KAFKA_TOPIC_NAME));
     topicName = props.getString(Config.KAFKA_TOPIC_NAME);
   }
 
+  public HashMap<String, Object> getKafkaProperties() {
+    final HashMap<String, Object> kafkaParams;
+    kafkaParams = new HashMap<String, Object>();
+    for (Object prop : props.keySet()) {
+      kafkaParams.put(prop.toString(), props.get(prop));
+    }
+    kafkaParams.put(ConsumerConfig.GROUP_ID_CONFIG, String.valueOf(new Random().nextInt(10000)));
+    kafkaParams.put("auto.offset.reset", "earliest");
+    kafkaParams.put("key.deserializer", StringDeserializer.class);
```

Comment:
if the user specified a different serializer via `props` this would override it? should we do a `putIfAbsent`